### PR TITLE
SPOTAUT-16833- ocean aws remaining apis

### PR DESCRIPTION
**Pull request by pripatra**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket

_Include a link to your Jira Ticket_
_Example: [JIRAISS-1234](https://spotinst.atlassian.net/browse/JIRAISS-1234)_
https://spotinst.atlassian.net/browse/SPOTAUT-16833
# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**)

### CODE REVIEW

File: spotinst_sdk2/models/ocean/aws/__init__.py [hunk 1]
```diff
@@ -848,6 +879,391 @@ def __init__(
...
+            instances_to_detach: List[str] = none,
+            should_decrement_target_capacity: bool = none,
+            should_terminate_instances: bool = none):
+        self.instances_to_detach = instances_to_detach
+        self.should_decrement_target_capacity = should_decrement_target_capacity
+        self.should_terminate_instances = should_terminate_instances
+# endregion
+    
+
+# region DetachInstances
```
Comment (anuragsharma-123) [inline spotinst_sdk2/models/ocean/aws/__init__.py]: region name is incorrect

File: spotinst_sdk2/clients/ocean/__init__.py [hunk 2]
```diff
@@ -460,8 +462,641 @@ def launch_nodes_in_vng(self, ocean_launch_spec_id: str, amount: int):
...
+        detach_instances (DetachInstances): Detach Instances Request
+
+        # Returns
+        (Object): Detach Instances API response
+        """
+        detach_instances_request = aws_ocean.DetachInstancesRequest(detach_instances)
+
+        excluded_missing_dict = self.exclude_missing(json.loads(detach_instances_request.toJSON()))
+
+        formatted_missing_dict = self.convert_json_with_list_of_lists(
```
Comment (anuragsharma-123) [inline spotinst_sdk2/clients/ocean/__init__.py]: convert_json_with_list_of_lists - This doesn't seem correct.

File: spotinst_sdk2/clients/ocean/__init__.py [hunk 3]
```diff
@@ -460,8 +462,641 @@ def launch_nodes_in_vng(self, ocean_launch_spec_id: str, amount: int):
...
+
+        return formatted_response["response"]["items"][0] 
+    
+    def get_migration_discovery(self, ocean_id: str, should_fetch_pods: bool):
+        """
+        Get information about nodes which can be migrated into Ocean.
+
+        # Arguments
+        ocean_id (String): ID of the Ocean Cluster
+        should_fetch_pods (bool): should_fetch_pods
```
Comment (anuragsharma-123) [inline spotinst_sdk2/clients/ocean/__init__.py]: Description should be English.

File: spotinst_sdk2/models/ocean/aws/__init__.py (hunk 1 above)
Comment (pripatra) [reply spotinst_sdk2/models/ocean/aws/__init__.py]: done

File: spotinst_sdk2/clients/ocean/__init__.py (hunk 3 above)
Comment (pripatra) [reply spotinst_sdk2/clients/ocean/__init__.py]: Changed

File: spotinst_sdk2/clients/ocean/__init__.py (hunk 2 above)
Comment (pripatra) [reply spotinst_sdk2/clients/ocean/__init__.py]: changed to convert_json

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**